packages/cdk8s-plus/src/service.ts

```diff
@@ -272,6 +272,37 @@ export class ServiceSpecDefinition {
     return [...this._ports];
   }
 
+  /**
+   * Exposes a deployment through this service.
+   *
+   * Requests will be routed to the port exposed by the first container in the
+   * deployment's pods. The label `cdk8s.deployment` will be added to pods in
+   * the deployment and used to select pods by this service.
+   *
+   * @param deployment The deployment to expose
+   * @param port The external port
+   */
+  public addDeployment(deployment: Deployment, port: number) {
+    const containers = deployment.spec.podSpecTemplate.containers;
+    if (containers.length === 0) {
+      throw new Error('Cannot expose a deployment without containers');
+    }
+
+    // create a label and attach it to the deployment pods
+    const selector = 'cdk8s.deployment';
+    const matcher = Names.toLabelValue(Node.of(deployment).path);
+
+    deployment.spec.podMetadataTemplate.addLabel(selector, matcher);
```

Comment:
I dont think this belongs here - why are pods only selected if the deployment is added to a service?

Reply:
Fixed.